iroha-cli/interactive/impl/interactive_transaction_cli.cpp

```diff
@@ -0,0 +1,443 @@
+    InteractiveTransactionCli::parseCreateAsset(std::string line) {
+      std::vector<std::string> notes = {"Full domain id"};
+      auto params = parseParams(line, CREATE_ASSET, notes);
+      if (not params.has_value()) {
+        return nullptr;
+      }
+      auto asset_name = params.value()[0];
+      auto domain_id = params.value()[1];
+      auto val = parser::toInt(params.value()[2]);
+      if (not val.has_value()) {
+        // error
+        return nullptr;
+      }
+      return generator_.generateCreateAsset(asset_name, domain_id, val.value());
+    }
+
+    std::shared_ptr<iroha::model::Command>
+    InteractiveTransactionCli::parseRemoveSignatory(std::string line) {
+      std::vector<std::string> notes = {"Account ID", "Public key to remove"};
+      auto params = parseParams(line, ADD_SIGN, notes);
+      if (not params.has_value()) {
+        return nullptr;
+      }
+      auto account_id = params.value()[0];
+      auto key = params.value()[1];
+      iroha::ed25519::pubkey_t pubkey;
+      iroha::hexstringToArray(key, pubkey);
+      return generator_.generateRemoveSignatory(account_id, pubkey);
+    }
+
+    std::shared_ptr<iroha::model::Command>
+    InteractiveTransactionCli::parseSetPermissions(std::string line) {
+      // TODO: implement when change permission model
+      return nullptr;
+    }
+
+    std::shared_ptr<iroha::model::Command>
+    InteractiveTransactionCli::parseSetQuorum(std::string line) {
+      std::vector<std::string> notes = {"Account id", "Quorum"};
+      auto params = parseParams(line, SET_QUO, notes);
+      if (not params.has_value()) {
+        return nullptr;
+      }
+      auto account_id = params.value()[0];
+      auto quorum = parser::toUint64(params.value()[2]);
+      if (not quorum.has_value()) {
+        std::cout << "Wrong format for quorum" << std::endl;
+        return nullptr;
+      }
+      return generator_.generateSetQuorum(account_id, quorum.value());
+    }
+    std::shared_ptr<iroha::model::Command>
+    InteractiveTransactionCli::parseSubtractAssetQuantity(std::string line) {
+      std::cout << "Not implemented" << std::endl;
+      // TODO: implement
+      return nullptr;
+    }
+
+    std::shared_ptr<iroha::model::Command>
+    InteractiveTransactionCli::parseTransferAsset(std::string line) {
+      std::vector<std::string> notes = {"Src account id", "Dest account id",
+                                        "Asset to transfer",
+                                        "Amount to transfer (integer part)",
+                                        "Amount to transfer (fractional part)"};
+
+      auto params = parseParams(line, TRAN_ASSET, notes);
+      if (not params.has_value()) {
+        return nullptr;
+      }
+      auto src_account_id = params.value()[0];
+      auto dest_account_id = params.value()[1];
+      auto asset_id = params.value()[2];
+      iroha::Amount amount;
+      auto val_int = parser::toUint64(params.value()[3]);
```

Comment:
This parsing also looks boilerplate. May be more clear to understanding would be move all error handling at the start of method?